Docs: add note and advice about .disabled and keyboard focus

patrickhlauke opened the pull request:
/cc @cvrebert

patrickhlauke committed:
Add note and advice about .disabled and keyboard focus

cvrebert commented on docs/components/pagination.md:
```diff
@@ -43,11 +43,18 @@ Simple pagination inspired by Rdio, great for apps and search results. The large
+{% callout warning %}
+#### Disabled links and keyboard access
+
+Although links styled as `.disabled` prevent mouse/touch click functionality, they will still be focusable with the keyboard. To prevent this, include `tabindex="-1"` on the link, or use a static, non-focusable element instead.
```
Given that we already say that  `pointer-events: none` isn't reliable with regard to browser compatibility, and might be removing it, I wouldn't say that they "prevent mouse/touch click functionality".

patrickhlauke replied:
If we were to remove `pointer-events:none`, then we should probably consider dropping the use of `<a href="..." class="disabled">` altogether and skip straight to swapping out the `<a>` with a `<span>` or similar, no?

patrickhlauke replied:
Incidentally, I've had a quick look through the docs, but can't find where we already say anything about `pointer-events:none`. Am I missing it somewhere?

cvrebert replied:
> skip straight to swapping out the `<a>` with a `<span>` or similar, no?

While we should definitely bring up `<span>`, I suspect there are cases where it won't be a feasible option so we should still cover `<a>`

patrickhlauke committed:
Tweak the wording for link functionality

to harmonise it with the callout given in pagination

patrickhlauke committed:
Remove Opera 18 and IE11 mention

As per our browser compat policy, we only claim support for latest
versions of Opera on desktop. Current Opera is version 34. From testing, IE11 does support `pointer-events:none` just fine now.

cvrebert commented on docs/components/pagination.md:
```diff
@@ -43,11 +43,18 @@ Simple pagination inspired by Rdio, great for apps and search results. The large
+{% callout warning %}
+#### Link functionality caveat
+
+The `.disabled` class uses `pointer-events: none` to try to disable the link functionality of `<a>`s, but that CSS property is not yet standardized and isn't fully supported in Internet Explorer 11\. In addition, even in browsers that do support `pointer-events: none`, keyboard navigation remains unaffected, meaning that sighted keyboard users and users of assistive technologies will still be abl ...
```
Extraneous backslash after "11"

cvrebert commented on docs/components/buttons.md:
```diff
@@ -125,7 +125,7 @@ As `<a>` elements don't support the `disabled` attribute, you must add the `.dis
 #### Link functionality caveat
 
-This class uses `pointer-events: none` to try to disable the link functionality of `<a>`s, but that CSS property is not yet standardized and isn't fully supported in Opera 18 and below, or in Internet Explorer 11\. In addition, even in browsers that do support `pointer-events: none`, keyboard navigation remains unaffected, meaning that sighted keyboard users and users of assistive technologies wi ...
+The `.disabled` class uses `pointer-events: none` to try to disable the link functionality of `<a>`s, but that CSS property is not yet standardized and isn't fully supported in Internet Explorer 11\. In addition, even in browsers that do support `pointer-events: none`, keyboard navigation remains unaffected, meaning that sighted keyboard users and users of assistive technologies will still be abl ...
```
Interestingly, caniuse now says that `pointer-events` is supported in IE11 (http://caniuse.com/#feat=pointer-events ), which seems to agree with my testing.

patrickhlauke replied:
Testing this it does indeed seem that IE11 supports `pointer-events:none` just fine. Removing that piece of advice.

patrickhlauke commented:
x-ref also https://github.com/twbs/bootstrap/pull/18699
